diff --git a/CMakeLists.txt b/CMakeLists.txt
index b56953c..1734b95 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -186,6 +186,10 @@ if(CMAKE_C_COMPILER_ID MATCHES "AppleClang|Clang|GNU|Intel" AND NOT CMAKE_SYSTEM
   endif()
 endif()
 
+if(SWITCH)
+  list(APPEND mi_cflags -DSWITCH)
+endif()
+
 if (MSVC AND MSVC_VERSION GREATER_EQUAL 1914)
   list(APPEND mi_cflags /Zc:__cplusplus)
 endif()
diff --git a/src/os.c b/src/os.c
index 8541523..04d531f 100644
--- a/src/os.c
+++ b/src/os.c
@@ -31,6 +31,9 @@ terms of the MIT license. A copy of the license can be found in the file
 #include <windows.h>
 #elif defined(__wasi__)
 // stdlib.h is all we need, and has already been included in mimalloc.h
+#elif defined(SWITCH)
+#include <malloc.h> // on switch, just use default malloc for now
+#include <unistd.h>    // sysconf
 #else
 #include <sys/mman.h>  // mmap
 #include <unistd.h>    // sysconf
@@ -213,6 +216,13 @@ void _mi_os_init() {
   os_page_size = 0x10000; // WebAssembly has a fixed page size: 64KB
   os_alloc_granularity = 16;
 }
+#elif defined(__SWITCH__)
+void _mi_os_init() {
+  // devkitPro's unistd declares sysconf but it is not implemented
+  // So just guess
+  os_page_size = 0x10000;
+  os_alloc_granularity = 16;
+}
 #else
 void _mi_os_init() {
   // get the page size
@@ -238,6 +248,8 @@ static bool mi_os_mem_free(void* addr, size_t size, bool was_committed, mi_stats
   err = (VirtualFree(addr, 0, MEM_RELEASE) == 0);
 #elif defined(__wasi__)
   err = 0; // WebAssembly's heap cannot be shrunk
+#elif defined(SWITCH)
+  free(addr);
 #else
   err = (munmap(addr, size) == -1);
 #endif
@@ -331,6 +343,8 @@ static void* mi_wasm_heap_grow(size_t size, size_t try_alignment) {
   }
   return (void*)aligned_base;
 }
+#elif defined(SWITCH)
+// Don't need anything here yet
 #else
 #define MI_OS_USE_MMAP
 static void* mi_unix_mmapx(void* addr, size_t size, size_t try_alignment, int protect_flags, int flags, int fd) {
@@ -539,6 +553,8 @@ static void* mi_os_mem_alloc(size_t size, size_t try_alignment, bool commit, boo
   #elif defined(__wasi__)
     *is_large = false;
     p = mi_wasm_heap_grow(size, try_alignment);
+  #elif defined(SWITCH)
+    p = malloc(size);
   #else
     int protect_flags = (commit ? (PROT_WRITE | PROT_READ) : PROT_NONE);
     p = mi_unix_mmap(NULL, size, try_alignment, protect_flags, false, allow_large, is_large);
@@ -734,6 +750,8 @@ static bool mi_os_commitx(void* addr, size_t size, bool commit, bool conservativ
   }
   #elif defined(__wasi__)
   // WebAssembly guests can't control memory protection
+  #elif defined(SWITCH)
+  // Switch guests also can't control memory protection (maybe?)
   #elif defined(MAP_FIXED)
   if (!commit) {
     // use mmap with MAP_FIXED to discard the existing memory (and reduce commit charge)
@@ -824,6 +842,8 @@ static bool mi_os_resetx(void* addr, size_t size, bool reset, mi_stats_t* stats)
   }
 #elif defined(__wasi__)
   int err = 0;
+#elif defined(SWITCH)
+  int err = 0;
 #else
   int err = madvise(start, csize, MADV_DONTNEED);
 #endif
@@ -882,6 +902,8 @@ static  bool mi_os_protectx(void* addr, size_t size, bool protect) {
   err = (ok ? 0 : GetLastError());
 #elif defined(__wasi__)
   err = 0;
+#elif defined(SWITCH)
+  err = 0;
 #else
   err = mprotect(start, csize, protect ? PROT_NONE : (PROT_READ | PROT_WRITE));
   if (err != 0) { err = errno; }
diff --git a/src/stats.c b/src/stats.c
index c94fbde..d5e3b98 100644
--- a/src/stats.c
+++ b/src/stats.c
@@ -503,7 +503,7 @@ static void mi_stat_process_info(mi_msecs_t* elapsed, mi_msecs_t* utime, mi_msec
   getrusage(RUSAGE_SELF, &rusage);
   *utime = timeval_secs(&rusage.ru_utime);
   *stime = timeval_secs(&rusage.ru_stime);
-#if !defined(__HAIKU__)
+#if !defined(__HAIKU__) && !defined(SWITCH)
   *page_faults = rusage.ru_majflt;
 #endif
   // estimate commit using our stats
@@ -533,7 +533,7 @@ static void mi_stat_process_info(mi_msecs_t* elapsed, mi_msecs_t* utime, mi_msec
 }
 
 #else
-#ifndef __wasi__
+#ifndef __wasi__ || SWITCH
 // WebAssembly instances are not processes
 #pragma message("define a way to get process info")
 #endif
