diff --git a/lib/basisu/encoder/basisu_enc.cpp b/lib/basisu/encoder/basisu_enc.cpp
index f02fb62c..c02e689a 100644
--- a/lib/basisu/encoder/basisu_enc.cpp
+++ b/lib/basisu/encoder/basisu_enc.cpp
@@ -208,7 +208,7 @@ namespace basisu
 		*pTicks = 1000000;
 	}
 #elif defined(__GNUC__)
-#include <sys/timex.h>
+#include <sys/time.h>
 	inline void query_counter(timer_ticks* pTicks)
 	{
 		struct timeval cur_time;
diff --git a/lib/gl_funcs.c b/lib/gl_funcs.c
index bce164c3..422a5f8f 100644
--- a/lib/gl_funcs.c
+++ b/lib/gl_funcs.c
@@ -49,6 +49,9 @@
 #if WINDOWS
 #define WINDOWS_LEAN_AND_MEAN
 #include <windows.h>
+#elif __SWITCH__
+// Do nothing
+#define WINAPI
 #else
 #include <dlfcn.h>
 #include <stdlib.h>
@@ -76,6 +79,9 @@ extern void* emscripten_GetProcAddress(const char *name_);
 #define LoadProcAddr(lib, proc) emscripten_GetProcAddress(proc)
 #define LIBRARY_NAME "unused"
 void* ktxOpenGLModuleHandle;
+#elif __SWITCH__
+// Do nothing; GL must be statically linked
+void* ktxOpenGLModuleHandle;
 #else
 #error "Don\'t know how to load symbols on this OS."
 #endif
@@ -88,6 +94,8 @@ static const char* noloadmsg = "Could not load OpenGL command: %s!\n";
 /* Define pointers for functions libktx is using. */
 struct glFuncPtrs gl;
 
+#if defined(KTX_USE_FUNCPTRS_FOR_OPENGL)
+
 #if defined(__GNUC__)
 // This strange casting is because dlsym returns a void* thus is not
 // compatible with ISO C which forbids conversion of object pointers
@@ -192,3 +200,14 @@ ktxLoadOpenGLLibrary(void)
 
 #undef GL_FUNCTION
 
+#else
+
+ktx_error_code_e
+ktxLoadOpenGLLibrary(void)
+{
+    return KTX_SUCCESS;
+}
+
+#endif
+
+
diff --git a/lib/gl_funcs.h b/lib/gl_funcs.h
index 0c8f6cce..98c54314 100644
--- a/lib/gl_funcs.h
+++ b/lib/gl_funcs.h
@@ -20,6 +20,12 @@
 #ifndef _GL_FUNCS_H_
 #define _GL_FUNCS_H_
 
+#ifdef __SWITCH__
+//#define KTX_USE_FUNCPTRS_FOR_OPENGL 0
+#elif !defined(KTX_USE_FUNCPTRS_FOR_OPENGL)
+#define KTX_USE_FUNCPTRS_FOR_OPENGL 1
+#endif
+
 #undef GL_GLEXT_PROTOTYPES // Just to be sure.
 #include "GL/glcorearb.h"
 #include "ktx.h"
diff --git a/lib/vk_funcs.h b/lib/vk_funcs.h
index b1c9c870..660b7715 100644
--- a/lib/vk_funcs.h
+++ b/lib/vk_funcs.h
@@ -21,7 +21,9 @@
 #ifndef _VK_FUNCS_H_
 #define _VK_FUNCS_H_
 
-#if !defined(KTX_USE_FUNCPTRS_FOR_VULKAN)
+#ifdef __SWITCH__
+//#define KTX_USE_FUNCPTRS_FOR_VULKAN 0
+#elif !defined(KTX_USE_FUNCPTRS_FOR_VULKAN)
 #define KTX_USE_FUNCPTRS_FOR_VULKAN 1
 #endif
 
